src/environment.cpp

```diff
@@ -112,10 +112,52 @@ inline static bool isPointableNode(const MapNode &n,
 	       (liquids_pointable && features.isLiquid());
 }
 
+inline static void updateMaxOverlap(f32 overlap, f32 extent, f32& max_overlap, f32& max_overlap_extent)
+{
+	if (overlap > 0 && (max_overlap_extent == 0 || overlap * max_overlap_extent > extent * max_overlap)) {
```

Comment:
Would you please be so nice to add a comment here to describe what the main idea of this calculation is, and what the variables indicate?

It seems to be some sort of area size comparison...

Reply:
Done